hub/api/dataset.py

```diff
@@ -23,7 +26,9 @@ def __init__(
         path: str = "",
         mode: str = "a",
         ds_slice: slice = slice(None),
-        provider: Optional[StorageProvider] = None,
+        memory_cache_size: int = DEFAULT_MEMORY_CACHE_SIZE,
+        local_cache_size: int = DEFAULT_LOCAL_CACHE_SIZE,
```

Comment:
is there any way we can remove these parameters, or move them to somewhere else? i don't think these should be dataset parameters, maybe we can do something like environment variables? this seems system-level.

Reply:
like for example, `HUB_MEMORY_CACHE_SIZE` and `HUB_LOCAL_CACHE_SIZE`, we always want datasets to be created using this.

and if you are worried about best practice, this is often how things are done (pytorch uses env variables for some of their cache stuff (https://pytorch.org/docs/stable/notes/cuda.html#memory-management) and im sure tensorflow does as well)

Reply:
this is very helpful for when we create docker containers also. that way a user never has to use this (unless they're super advanced)

Reply:
I disagree, these should definitely be dataset level and not system level. Depending on what your usecase is, you might want to use a full dataset size local cache for one dataset, and for another small dataset you might want to use just a memory cache and no local cache. A user could even have both these datasets open at once.

Reply:
yeah, but it shouldn't be on the burden of the user to set these values...

Reply:
we should set these values for them and not expose them directly in the API. this is going to be very rare use cases, and we should reduce the parameter clutter